### PR TITLE
docs: Relicense from AGPL 3 to Apache 2

### PR DESCRIPTION
**Pull request by timmc-edx**
See discussion on https://github.com/openedx/tcril-engineering/issues/289 for background (and authorization).
**Merge checklist:**
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed

### CODE REVIEW

File: LICENSE.txt [hunk 1]
```diff
@@ -1,674 +1,202 @@
...
-of the code.  There are many ways you could offer source, and different
-solutions will be better for different programs; see section 13 for the
-specific requirements.
-
-  You should also get your employer (if you work as a programmer) or school,
-if any, to sign a "copyright disclaimer" for the program, if necessary.
-For more information on this, and how to apply and follow the GNU AGPL, see
-<http://www.gnu.org/licenses/>.
-
-EdX Inc. wishes to state, in clarification of the above license terms, that
```
Comment (timmc-edx) [inline LICENSE.txt:664]: Of possible interest, note the custom addendum we had here. (Nothing similar should be required for Apache 2, as far as I am aware.)

File: setup.py [hunk 2]
```diff
@@ -121,15 +121,14 @@ def is_requirement(line):
             'log-sensitive = edx_django_utils.logging.internal.log_sensitive:cli',
         ],
     },
-    license="AGPL 3.0",
```
Comment (timmc-edx) [inline setup.py:124]: Apparently this field is only intended for use when you don't have a license with a known classifier, which we do in fact supply farther down.

File: LICENSE.txt (hunk 1 above)
Comment (kdmccormick) [reply LICENSE.txt:664]: Thanks for pointing that out, I'd never noticed that addendum. I agree that it doesn't matter for this PR, but it's good to know about.

**Review by kdmccormick — APPROVED** [review]
Looks good, thanks a bunch @timmc-edx .

**Pull request merged**